llvm/lib/Transforms/Utils/ValueMapper.cpp

```diff
@@ -552,6 +552,7 @@ void Mapper::remapDPValue(DPValue &V) {
       V.setKillAddress();
     else if (NewAddr)
       V.setAddress(NewAddr);
+    V.setAssignId(cast<DIAssignID>(mapMetadata(V.getAssignID())));
```

Comment:
🤦 Surprised this didn't show up in any testing earlier, thanks for spotting it quickly though!